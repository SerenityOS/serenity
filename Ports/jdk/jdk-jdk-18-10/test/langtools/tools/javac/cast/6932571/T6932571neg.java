/*
 * @test    /nodynamiccopyright/
 * @bug     6932571
 * @summary Compiling Generics causing Inconvertible types
 * @compile/fail/ref=T6932571neg.out -XDrawDiagnostics T6932571neg.java
 */

class T6932571neg {
    interface I<T>{ }
    interface I1 extends I<String> {}
    static class Y implements I<String> {}
    final static class S implements I<String> {}

    <G extends I<G>> void test() {
        S s = new S();
        G g = (G) s;
    }
}
