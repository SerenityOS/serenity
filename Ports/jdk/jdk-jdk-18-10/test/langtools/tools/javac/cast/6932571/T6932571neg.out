T6932571neg.java:16:19: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: T6932571neg.S, G)
1 error
